compiler/luci-interpreter/src/kernels/Add.test.cpp

```diff
@@ -27,57 +27,106 @@ namespace
-      {0.0f, 2.6f, 0.5f, 1.0f, 1.8f, 0.0f, 1.4f, 3.1f, 0.0f, 0.5f, 1.7f, 0.0f},
-      {0.0f, 2.5f, 0.0f, 2.6f, 0.0f, 1.9f, 1.1f, 0.7f, 1.2f, 0.8f, 0.5f, 0.1f,
-       1.0f, 0.0f, 1.1f, 0.0f, 0.4f, 0.0f, 1.7f, 3.3f, 2.2f, 3.8f, 2.1f, 3.7f,
-       0.0f, 0.5f, 0.0f, 1.0f, 0.0f, 0.9f, 1.2f, 0.0f, 1.7f, 0.0f, 1.6f, 0.0f},
-      {0.0f, 2.5f, 1.2f, 0.8f, 0.4f, 0.0f, 1.7f, 3.3f, 0.0f, 1.0f, 1.6f, 0.0f}};
-  std::vector<float> input1_data{-0.3f, 2.3f, 0.9f,  0.5f, 0.8f, -1.1f,
-                                 1.2f,  2.8f, -1.6f, 0.0f, 0.7f, -2.2f};
-  std::vector<float> input2_data{0.2f, 0.3f, -0.4f, 0.5f, 1.0f, 0.9f};
-  for (size_t i = 0; i < test_shapes.size(); ++i)
+  float kQuantizedStep = (max - min) / 255.0;
+  return kQuantizedStep;
+}
+
+template <typename T>
+void Check(std::initializer_list<int32_t> input1_shape, std::initializer_list<int32_t> input2_shape,
+           std::initializer_list<int32_t> output_shape, std::initializer_list<float> input1_data,
+           std::initializer_list<float> input2_data, std::initializer_list<float> output_data)
+{
+  float min, max;
+  min =
+      std::min(input1_data) > std::min(input2_data) ? std::min(input2_data) : std::min(input1_data);
+  max =
+      std::max(input1_data) > std::max(input2_data) ? std::max(input1_data) : std::max(input2_data);
+  float kQuantizedTolerance = GetTolerance(min, max);
+  std::pair<float, int32_t> input_quant_param = quantizationParams<T>(min, max);
+  Tensor input1_tensor{getElementType<T>(),
+                       input1_shape,
+                       {{input_quant_param.first}, {input_quant_param.second}},
+                       ""};
+  Tensor input2_tensor{getElementType<T>(),
+                       input2_shape,
+                       {{input_quant_param.first}, {input_quant_param.second}},
+                       ""};
+  if (std::is_floating_point<T>::value)
   {
-    Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>(base_shape, input1_data);
-    Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>(test_shapes[i], input2_data);
-    Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+    input1_tensor.writeData(input1_data.begin(), input1_data.size() * sizeof(T));
+    input2_tensor.writeData(input2_data.begin(), input2_data.size() * sizeof(T));
+  }
+  else
+  {
+    std::vector<T> quantized_input1_value =
+        quantize<T>(input1_data, input_quant_param.first, input_quant_param.second);
+    std::vector<T> quantized_input2_value =
+        quantize<T>(input2_data, input_quant_param.first, input_quant_param.second);
+    input1_tensor.writeData(quantized_input1_value.data(),
+                            quantized_input1_value.size() * sizeof(T));
+    input2_tensor.writeData(quantized_input2_value.data(),
+                            quantized_input2_value.size() * sizeof(T));
+  }
+  std::pair<float, int32_t> output_quant_param = quantizationParams<T>(0, max * 2);
```

Comment:
How did you get (0, max *2) ?

Reply:
I just thought that summation of to tensor is below max*2 and 0 is minimal value of Relu function.

but, I thought sum of min/max value in each tensor would be better.